www/js/model.js

```diff
@@ -243,6 +243,25 @@ define([
 						}, this);
 				return flag ? true : false;
 			},
+			equationNodesMatchesGivenSolution: function(){
```

Comment:
This condition doesn't match the spec in #464.  It is checking if every author equation has been added to the student model and if all of them are complete.  It was just supposed to check if the student model has the right number of every schema.  So that created two bugs:

1) The user must fill in the entities and variables before the check will work.  (Fixed by only counting schemas, ignore entities and variables)
2) If the student has one or more incomplete equations , the check fails even if all the equations are present.  (Fixed by removing isComplete check.)  

However the second bug is also in the 464 design.  If the user clicks "Add Equation" a bunch of times without setting the schema, they can add more than the max number of equations and end up in the same situation that 464 was trying to fix.  So instead of using the old design I wrote in #464, make the following changes in this pull request:

1. Rename the function to `checkEquationNodesDone` to `hasEnoughEquationNodes`
1. Change that function to just count the number of equations in the Author's model and count the number of equations in the student model.  If the student's count is >= the Author's count, show the message and don't add the equation.  If the counting function exists already, do reuse it.